### PR TITLE
`IsogonalOctagon` composite surface

### PR DESCRIPTION
**Pull request by yardasol**
This PR adds a new `CompositeSurface` class, `IsogonalOctagon`, and associated unit tests. This PR also adds the `IsogonalOctagon` class to the Sphinx Docs.

### CODE REVIEW

**Comment by paulromano** [comment]
Thanks @yardasol! Will take a look at this PR and your other one within the next few days.

File: openmc/model/surface_composite.py
Comment (paulromano) [inline openmc/model/surface_composite.py:128]: Is there any situation where we would actually want to allow this? If not we should just make these exceptions rather than warnings.

File: openmc/model/surface_composite.py
Comment (paulromano) [inline openmc/model/surface_composite.py:169]:
```suggestion
        for p in points:
            p[:] = p[coord_map]
```

File: openmc/model/surface_composite.py
Comment (paulromano) [inline openmc/model/surface_composite.py:185]: If you were to change `coord_map` for the y-axis case to `[1, 2, 0]`, would that prevent us having to use the `if` block here?

File: tests/unit_tests/test_surface_composite.py
Comment (paulromano) [inline tests/unit_tests/test_surface_composite.py:224]:
```suggestion
```

File: openmc/model/surface_composite.py
Comment (paulromano) [inline openmc/model/surface_composite.py:162]:
```suggestion
        if axis == 'z':
            coord_map = [0, 1, 2]
            self.top = openmc.YPlane(ctop, **kwargs)
            self.bottom = openmc.YPlane(cbottom, **kwargs)
            self.right = openmc.XPlane(cright, **kwargs)
            self.left = openmc.XPlane(cleft, **kwargs)
        elif axis == 'y':
            coord_map = [0, 2, 1]
            self.top = openmc.ZPlane(ctop, **kwargs)
            self.bottom = openmc.ZPlane(cbottom, **kwargs)
            self.right = openmc.XPlane(cright, **kwargs)
            self.left = openmc.XPlane(cleft, **kwargs)
        elif axis == 'x':
            coord_map = [2, 0, 1]
            self.top = openmc.ZPlane(ctop, **kwargs)
            self.bottom = openmc.ZPlane(cbottom, **kwargs)
            self.right = openmc.YPlane(cright, **kwargs)
            self.left = openmc.YPlane(cleft, **kwargs)
```

File: openmc/model/surface_composite.py
Comment (paulromano) [inline openmc/model/surface_composite.py:179]:
```suggestion
```

File: openmc/model/surface_composite.py
Comment (paulromano) [inline openmc/model/surface_composite.py:55]:
```suggestion

class IsogonalOctagon(CompositeSurface):
```

File: tests/unit_tests/test_surface_composite.py
Comment (paulromano) [inline tests/unit_tests/test_surface_composite.py:165]:
```suggestion
    center = np.array([0., 0.])
    point_pos = np.array([0.8, 0.8])
    point_neg = np.array([0.7, 0.7])
```

**Review by paulromano — CHANGES_REQUESTED** [review]
Nice work @yardasol! A few suggestions for improvements are below, but once those are addressed we can get this merged.

**Comment by yardasol** [comment]
@paulromano I made the requested changes and updates the docstrings and unit tests for consistency.

File: openmc/model/surface_composite.py
Comment (yardasol) [inline openmc/model/surface_composite.py:112]:
```suggestion
```

File: openmc/model/surface_composite.py
Comment (paulromano) [inline openmc/model/surface_composite.py:4]: For consistency with the rest of the codebase:
```suggestion

import numpy as np
```

File: openmc/model/surface_composite.py
Comment (paulromano) [inline openmc/model/surface_composite.py:93]:
```suggestion
    right : openmc.YPlane, openmc.ZPlane, or openmc.XPlane
        Right planar surface of octagon
    left : openmc.YPlane, openmc.ZPlane, or openmc.XPlane
```

File: openmc/model/surface_composite.py
Comment (paulromano) [inline openmc/model/surface_composite.py:130]:
```suggestion
```

File: openmc/model/surface_composite.py
Comment (paulromano) [inline openmc/model/surface_composite.py:139]: If we're always using `L_prep_ax1/2` might as well just have the variable be half of what it is now (eliminating all the `/2`)

File: tests/unit_tests/test_surface_composite.py
Comment (paulromano) [inline tests/unit_tests/test_surface_composite.py:194]:
```suggestion
    xmax, ymax, zmax = center + r1
```

File: tests/unit_tests/test_surface_composite.py
Comment (paulromano) [inline tests/unit_tests/test_surface_composite.py:197]:
```suggestion
    xmin, ymin, zmin = coord_min
```

File: tests/unit_tests/test_surface_composite.py
Comment (paulromano) [inline tests/unit_tests/test_surface_composite.py:220]:
```suggestion
    # Check invalid r1, r2 combinations
    with pytest.raises(ValueError):
        openmc.model.IsogonalOctagon(center, r1=1.0, r2=10.)
    with pytest.raises(ValueError):
        openmc.model.IsogonalOctagon(center, r1=10.0, r2=1.)
    
    # Make sure repr works
```

File: openmc/model/surface_composite.py
Comment (paulromano) [inline openmc/model/surface_composite.py:139]:
```suggestion
        p1_ur = np.array([L_perp_ax1/2, r1, 0.])
        p2_ur = np.array([r1, L_perp_ax1/2, 0.])
        p3_ur = np.array([r1, L_perp_ax1/2, 1.])

        p1_lr = np.array([r1, -L_perp_ax1/2, 0.])
        p2_lr = np.array([L_perp_ax1/2, -r1, 0.])
        p3_lr = np.array([L_perp_ax1/2, -r1, 1.])
```

**Review by paulromano — CHANGES_REQUESTED** [review]
A few more small requests:

File: openmc/model/surface_composite.py
Comment (yardasol) [reply openmc/model/surface_composite.py:93]: nice catch!

File: openmc/model/surface_composite.py
Comment (yardasol) [reply openmc/model/surface_composite.py:130]: I had forgot I had that line in there. It never gets used so should be removed.

**Comment by yardasol** [comment]
Good suggestions @paulromano. I've applied them.

**Review by paulromano — APPROVED** [review]
Good to go! Will merge once CI gives the green light

**Pull request merged**
